### PR TITLE
look up brand articles file by path param

### PR DESCRIPTION
**Pull request by srilq**
#### This resolves Clubhouse issue: [#9925](https://app.clubhouse.io/condenastinternational/story/9925/allow-demo-api-to-serve-different-articles-for-different-brands)

- [x] Tests have been written
- [x] Checked that the new code runs properly

Changes route from `/api/articles/:id` to `/:brand/articles/:id`.

Paired with @almostagile.

### CODE REVIEW

File: src/controllers/getArticle.spec.js [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+const defaultArticles = require('../data/articles');
+
+const brandArticles = {
+  dmo: defaultArticles
+};
+const path = `${__dirname}/../data/`;
+fs.readdirSync(path)
+  .forEach((file) => {
+    const brandMatch = file.match(/^articles-(\S+)\.json/);
+    if (!brandMatch) return;
```
Comment (srilq) [inline src/controllers/getArticle.spec.js]: Maybe should be `brandMatch.length < 2`?

File: src/controllers/getArticle.spec.js [hunk 2]
```diff
@@ -26,31 +38,60 @@ describe('GET /api/articles/:id', () => {
...
+    await server.ready();
+
+    const { id } = defaultArticles[0];
+    const res = await server.inject({ method: 'GET', url: `/${brand}/articles/${id}` });
+
+    expect(res.statusCode).toBe(200);
+
+    const body = JSON.parse(res.payload);
+
+    expect(body.brand).toBe(brand);
```
Comment (srilq) [inline src/controllers/getArticle.spec.js]: Should assert some property from `defaultArticles[0]` such as `defaultArticles[0].id` in response.

File: src/controllers/getArticle.spec.js [hunk 3]
```diff
@@ -26,31 +38,60 @@ describe('GET /api/articles/:id', () => {
...
-    const res = await server.inject({ method: 'GET', url: `/api/articles/${id}?brand=${brandOverride}` });
+    const res = await server.inject({ method: 'GET', url: `/${brand}/articles/${id}` });
 
     expect(res.statusCode).toBe(200);
 
     const body = JSON.parse(res.payload);
 
-    expect(body.brand).toBe(brandOverride);
+    expect(body.id).toBe(id);
+    expect(body.publishHistory.uri).toBe('mode/test-flyway');
```
Comment (srilq) [inline src/controllers/getArticle.spec.js]: Could look up id and uri in `brandArticles`.

File: src/controllers/getArticle.spec.js (hunk 1 above)
Comment (almostagile) [reply src/controllers/getArticle.spec.js]: No because `match()` return `null` if no match

Comment (srilq) [reply src/controllers/getArticle.spec.js]: Right yes, because the capture isn't optional. 👍

File: src/controllers/getArticle.spec.js (hunk 2 above)
Comment (almostagile) [reply src/controllers/getArticle.spec.js]: Yes that would make sense

File: src/controllers/getArticle.spec.js (hunk 3 above)
Comment (almostagile) [reply src/controllers/getArticle.spec.js]: Oh yes we could now

**Comment by almostagile** [comment]
Finally ready for review by someone else. I fixed the pagination bug at last.

**Comment by srilq** [comment]
@almostagile We should update the Integration API docs to reflect changes here.

**Pull request merged**